### PR TITLE
Apex component naming proposal

### LINKED ISSUE #441: Consistent naming for apex stack components
### Describe the Problem Statement

Currently we are are using multiple names for different components across our documentation and code, such as

* Apex Stack is referred as Apex Service, Apex ApiServer, Apex Controller, Apex Control Plane, Apex SaaS
* Apex Agent is referred as hub-router, relay, agent, discovery node

It would be really helpful to write a coherent documentation and code if we can zero down on the naming of these components.

### PR DESCRIPTION
**Pull request by vishnoianil**
Initial proposal to start the discussion around finding consistent naming of the apex components, so that we can write coherent document/code

Resolves #441

### CODE REVIEW

**Comment by dave-tucker** [comment]
Can we combine this with #440 and perhaps have a `docs/naming.md`

File: docs/proposals/naming.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  * Apex backend-cli
+  * Apex backend-web
+* Apex Agent: Agent running on each node, just doing Agent's job.
+  * Binary Naming: We need to go one more level down to clarify the binary naming as we are planning to separate the relay functionality out of agent.
+    * apexd: Agent binary (running as a daemon) that does the magic of connectivity.
+    * apex: A command line utility to do read/write operation through Apexd
+* Apex ICE: A node running the discovery and relay function.
+  * Binary Naming:
+    * apexiced: Binary (running as a daemon) that does the magic of discovery and relay.
+    * apexice: A command line utility to do read/write operation through Apexiced
```
Comment (chirino) [inline docs/proposals/naming.md:35]: ICE is the protocol the client uses to discover, prioritize, and test the endpoint candidate addresses of peers.  I don't really like it to describe a node that would be running stun and relay functionality.  I would rather call it something like relay.

**Comment by russellb** [comment]
> Can we combine this with #440 and perhaps have a `docs/naming.md`

@vishnoianil I will incorporate this into my PR if you're good with it.

**Comment by vishnoianil** [comment]
> > Can we combine this with #440 and perhaps have a `docs/naming.md`
> 
> @vishnoianil I will incorporate this into my PR if you're good with it.

Yup, sounds good.

File: docs/proposals/naming.md (hunk 1 above)
Comment (vishnoianil) [reply docs/proposals/naming.md:35]: yeah makes sense. Relay was an option i was also thinking, but again it's kind of restrict to one functionality, and as of now it's doing more than that. Let's continue discussion on #440

**Comment by vishnoianil** [comment]
Proposal merged with #440

**Pull request closed without merging**